homeassistant/components/rainbird/switch.py

```diff
@@ -64,6 +73,18 @@ def start_irrigation(service):
         schema=SERVICE_SCHEMA_IRRIGATION,
     )
 
+    def delay_irrigation(service):
+        days = service.data[ATTR_DAYS]
+
+        controller.set_rain_delay(days=days)
+
+    hass.services.register(
```

Comment:
If the service is not specific to this switch platform, please move the registration to the integration init module instead.

Reply:
Done! I moved in the setup ... 
I don't like the way I collect the `controller` in the code ... I would prefer to get it from `_setup_controller`.